src/components/data-management/UploadDetailsModal.jsx

```diff
@@ -12,20 +12,19 @@ dayjs.extend(utc);
 
 const UploadDetailsModal = (props) => {
   const {
-    onCancel, file, extraFields, onDownload, onRetry, onDelete,
+    onCancel, data, extraFields, onDownload, onRetry, onDelete,
   } = props;
 
   const {
-    name, upload, size, lastModified, fileObject = undefined,
-  } = file ?? {};
+    upload, size, lastModified, fileObject = undefined,
+  } = data ?? {};
```

Comment:
considering that `data` is a required property, and it's initialized in `SamplesTableCell` like `const [uploadDetailsModalData, setUploadDetailsModalData] = useState(false);`. 

Wouldn't it be easier to initialize it to `{}` and get rid of the data **?? {};** ?

Reply:
I checked the 2 uses of `UploadDetailsModal` and think it can't be false. Also the line here is using a `??` but then assumes that `upload`, `size`, `lastModified` are defined. I just removed the `?? {}` instead.

Reply:
As an aside, I've noticed a lot of these type-switching variables in the UI: 
`const { progress, status } = upload ?? false`

I don't think we should allow variables to switch between bools and objects like it happens here. We should use `null` or `undefined` if we want to indicate a variable is not set.

I am bringing this up so you can look out for this too in future reviews)

Reply:
sounds good